packages/plugingen/src/utils/asyncPipe.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Utility function that allows for piping a series of async functions together
+ *
+ * @param fns - functions to pipe
+ * @returns result of the pipe
+ */
+export const asyncPipe =
+  <S, O, F>(...fns: ((s: S, o?: O, f?: F) => Promise<S>)[]) =>
+  async (s: S, o?: O, f?: F) => {
+    let result = s;
+    for (const fn of fns) {
+      result = await fn(result, o, f);
+    }
+    return result;
+  };
```

Comment:
ditto: aa-core or aa-utils?

Reply:
oh it's already in aa-core. I just need to import it here

Reply:
OH I didn't want this package to depend on core actually

Reply:
nm I already depend on aa-accounts so may as well I guess